CONTRIBUTING.md

```diff
@@ -1,25 +1,74 @@
-# How to contribute to Delta Kernel Rust
+# Contributing to Delta Kernel (Rust)
 
-Welcome!  We'd love to have you contribute to Delta Kernel Rust!
+> [!NOTE]
+> **Found a bug?** Please first search [existing issues] to avoid duplicates. If you find a related
+> issue, add your details there. Otherwise, open a new issue with a reproducible example including
+> Rust version, delta-kernel-rs version, code executed, and error message.
 
-## Did you find a bug?
+[existing issues]: (https://github.com/delta-io/delta-kernel-rs/issues)
 
-Create an issue with a reproducible example.  Please specify the Rust version, delta-kernel-rs version, the code executed, and the error message.
+## How to Contribute
 
-## Did you create a PR to fix a bug?
+We follow a structured contribution process to ensure high-quality code:
 
-Open a pull request and add "Fixes #issue_number" in the PR description.
+1. **Start with an issue and/or design sketch**: Open an issue describing what you want to
+   contribute and why. Continue to step 2 after reaching some consensus. This helps us avoid wasted
+   effort (perhaps you were building something that someone else was already pursuing or already
+   explored and rejected). Including a design sketch will help drive consensus (often a simple
+   diagram or bullet points outlining high-level changes is sufficient).
+2. **Prototype/POC**: Create a PR marked as prototype/draft (not intended to merge) and gather
+   feedback to further derisk the design. This PR is not intended to be merged but will guide the
+   implementation and serve as a proving ground for the design. Then, pieces are torn out into
+   smaller PRs that can be merged.
```

Comment:
Let's give em something to target :)
```suggestion
   smaller PRs (1-200 LOC) that can be merged.
```

Reply:
hm fair point though I hesitate to put numbers in here since "small" for some things might be 600 lines (100 lines material change + 500 lines tests)

how about I merge this and we can take this up as an improvement if others feel it's useful?